dot/parachain/prospective-parachains/prospective_parachains_test.go

```diff
@@ -1025,3 +1053,135 @@ func TestHandleBacked(
 
 	require.Equal(t, hashes[0], parachaintypes.CandidateHash{Value: hash})
 }
+
+func TestActivateLeafSignalHandler(t *testing.T) {
+	cases := map[string]struct {
+		setup func(*testing.T) (*ProspectiveParachains, parachaintypes.ActiveLeavesUpdateSignal)
+	}{
+		"no_ancestry_and_no_previous_fragment_chains": {
```

Comment:
Did you use a table-based test because you expect more test cases to be added in the future?

Reply:
Yeah, that will be part of the integration test issue